client/src/index.tsx

```diff
@@ -10,6 +10,87 @@ import { GleanProvider } from "./telemetry/glean-context";
 
 // import * as serviceWorker from './serviceWorker';
 
+let attemptedFallback = false;
+window.addEventListener("error", async (error) => {
+  const root = document.querySelector("#root");
+  if (
+    !attemptedFallback &&
+    root &&
+    // only fall back if react gave up and unmounted, leaving an empty page
+    root.innerHTML.trim() === ""
+  ) {
+    attemptedFallback = true;
+    try {
+      const res = await fetch(window.location.pathname, {
+        cache: "force-cache",
+      });
+      const page = await res.text();
+      const parser = new DOMParser();
+      const fallbackRoot = parser
+        .parseFromString(page, "text/html")
+        .querySelector("#root");
+      if (fallbackRoot) {
+        root.innerHTML = fallbackRoot.innerHTML;
+        console.warn(
+          "Fell back to static HTML page due to unhandled error,",
+          "please report an issue on GitHub:",
+          generateGithubIssueLink(error.error)
+        );
+      } else {
+        throw new Error("no root");
+      }
+    } catch (fallbackError) {
+      console.warn(
+        "Failed falling back to static HTML page:",
+        fallbackError,
+        "please report an issue on GitHub:",
+        generateGithubIssueLink(error.error, fallbackError)
+      );
+    }
+  }
+  // Unlike other events, the error event is canceled by returning true from the handler instead of returning false.
+  // https://developer.mozilla.org/en-US/docs/Web/API/Window/error_event#usage_notes
+  return false;
+});
+
+function generateGithubIssueLink(error: Error, fallbackError?: any) {
+  try {
+    const params = new URLSearchParams();
+    params.append("title", `Unhandled error: ${error.message}`);
+    params.append("labels", "🐛 bug");
+    params.append(
+      "body",
```

Comment:
@caugner I reckon you might be able to improve upon this title, might want some different labels, or want to add some instructions to users

Reply:
No, I think it's good, we can always improve it later.